helper/predeployment/predeployment.go

```diff
@@ -0,0 +1,254 @@
+package predeployment
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/0xPolygon/polygon-edge/chain"
+	"github.com/0xPolygon/polygon-edge/contracts/staking"
+	"github.com/0xPolygon/polygon-edge/helper/hex"
+	"github.com/0xPolygon/polygon-edge/state"
+	itrie "github.com/0xPolygon/polygon-edge/state/immutable-trie"
+	"github.com/0xPolygon/polygon-edge/state/runtime"
+	"github.com/0xPolygon/polygon-edge/state/runtime/evm"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/umbracle/ethgo/abi"
+	"io/ioutil"
+	"math"
+	"math/big"
+	"os"
+	"strings"
+)
+
+var (
+	errABINotFound              = errors.New("abi field not found in specified JSON")
+	errBytecodeNotFound         = errors.New("bytecode field not found in specified JSON")
+	errDeployedBytecodeNotFound = errors.New("deployed bytecode field not found in specified JSON")
+)
+
+type contractArtifact struct {
+	ABI              []byte // the ABI of the Smart Contract
+	Bytecode         []byte // the raw bytecode of the Smart Contract
+	DeployedBytecode []byte // the deployed bytecode of the Smart Contract
+}
+
+// generateContractArtifact generates contract artifacts based on the
+// passed in Smart Contract JSON ABI
+func generateContractArtifact(filepath string) (*contractArtifact, error) {
+	var (
+		artifact *contractArtifact
+		err      error
+	)
+
+	// Read from the ABI from the JSON file
+	jsonFile, err := os.Open(filepath)
+	if err != nil {
+		return artifact, err
+	}
+
+	jsonRaw, err := ioutil.ReadAll(jsonFile)
+	if err != nil {
+		return artifact, err
+	}
+
+	// Fill out the fields in the JSON file
+	var jsonResult map[string]interface{}
+	if err = json.Unmarshal(jsonRaw, &jsonResult); err != nil {
+		return artifact, err
+	}
+
+	// Parse the ABI
+	abiRaw, ok := jsonResult["abi"]
+	if !ok {
+		return nil, errABINotFound
+	}
+
+	abiBytes, err := json.Marshal(abiRaw)
+	if err != nil {
+		return nil, fmt.Errorf("unable to marshal ABI to JSON, %w", err)
+	}
+
+	// Parse the bytecode
+	bytecode, ok := jsonResult["bytecode"].(string)
+	if !ok {
+		return nil, errBytecodeNotFound
+	}
+
+	hexBytecode, err := hex.DecodeString(strings.TrimPrefix(bytecode, "0x"))
+	if err != nil {
+		return nil, fmt.Errorf("unable to decode bytecode, %w", err)
+	}
+
+	// Parse deployed bytecode
+	deployedBytecode, ok := jsonResult["deployedBytecode"].(string)
```

Comment:
Shall we define key as a constant?

Reply:
Defined:

[248e378](https://github.com/0xPolygon/polygon-edge/pull/536/commits/248e3782e5eda590309e541d48fe5803166b61fa)